packages/manager/core/application/src/hooks/useFeatureAvailability.ts

```diff
@@ -1,47 +1,46 @@
-import { useQuery } from '@tanstack/react-query';
 import { apiClient } from '@ovh-ux/manager-core-api';
+import { useQuery } from '@tanstack/react-query';
 
-export const featureAvailabilityBaseQueryKey = 'feature-availability-';
-
-export type UseFeatureAvailabilityParams = {
-  parent: string;
-  featureList?: string[];
-};
+/**
+ * exemples
+ * const featureList = ['billing', 'webooo', 'web:microsoft'];
 
-const getFeaturesListToCheck = ({
-  parent,
-  featureList = [],
-}: UseFeatureAvailabilityParams) =>
-  [parent].concat(featureList.map((name) => `${parent}:${name}`));
+  const { data } = useFeatureAvailability(featureList);
 
-/**
- * @example
- * const {data, error, loading} = useFeatureAvailability(
- *   {parent: 'dedicated-nasha', featureList: ['feature1', 'feature2']},
- *   {parent: 'app2'},
- *   {parent: 'app3', featureList: ['feat1']}
- * );
- * const isAppAvailable = data.['dedicated-nasha'];
- * const isFeature1Available = data.feature1;
- * const isFeature2Available = data.feature2;
- * const isApp2Available = data.app2;
- * const isApp3Available = data.app3;
- * const isApp3Feature1Available = data.feat1;
+  const isBillingAvailable = data?.features.billing;
+  const isWebooooAvailable = data?.features.webooo;
+  const isMiscosoftAvailable = data?.features?.['web:microsoft'];
+ *
  */
-export const useFeatureAvailability = (
-  ...featureList: UseFeatureAvailabilityParams[]
-) => {
-  const featuresListToCheck = featureList.flatMap(getFeaturesListToCheck);
 
+interface FeatureAvailabilityResult {
+  features: Record<string, boolean>;
+}
+
+export const useFeatureAvailability = (
+  featureList: string[],
+): {
+  data: FeatureAvailabilityResult | undefined;
+} => {
   const fetchFeatureAvailabilityData = async () => {
     const result = await apiClient.aapi.get(
-      `/feature/${featuresListToCheck.join(',')}/availability`,
+      `/feature/${featureList.join(',')}/availability`,
     );
-    return result.data || {};
+
+    const featureAvailabilityResult: FeatureAvailabilityResult = {
+      features: {},
+    };
+
+    featureList.forEach((feature) => {
+      featureAvailabilityResult.features[feature] =
+        result.data[feature] || false;
+    });
+
+    return featureAvailabilityResult;
   };
 
-  return useQuery<Record<string, boolean>>(
-    [`${featureAvailabilityBaseQueryKey}-${featuresListToCheck.join('-')}`],
+  return useQuery<FeatureAvailabilityResult>(
+    [featureList.join('-')],
     fetchFeatureAvailabilityData,
-  );
+  ) as { data: FeatureAvailabilityResult };
```

Comment:
Use the object syntax ({ queryKey: [featureList.join('-')], queryFn: fetchFeature... })

And don't change the type, useQuery already gives a better type